File: src/test/java/com/example/travelator/abstractinsteadcomposition/java/BarTest.java
package com.example.travelator.abstractinsteadcomposition.java;

import org.junit.jupiter.api.Test;

import static org.assertj.core.api.AssertionsForClassTypes.assertThat;

class BarTest {
	
	@Test
	void testCase1() {
		FooImpl foo = new FooImpl();
		Bar bar = new Bar(foo);
		bar.print();
		assertThat(bar.message()).isEqualTo("Message of Bar");
		
	}
}

File: src/test/java/com/example/travelator/abstractinsteadcomposition/kotlin/BarTest.kt
package com.example.travelator.abstractinsteadcomposition.kotlin

import org.junit.jupiter.api.Test

class BarTest {

    @Test
    fun testCase1() {
        val foo = FooImpl()
        val bar = Bar(foo)
        bar.print()
        println(bar.message)
    }
}